### PR TITLE
Add tests to web app

### PR DESCRIPTION
**Pull request by Pasarus**
Needs to be merged after https://github.com/vais-ral/CILViewer/pull/270

adds tests for the trame based web app using the UnitTest python framework

### CODE REVIEW

File: Wrappers/Python/ccpi/viewer/CILViewer.py [hunk 1]
```diff
@@ -22,6 +22,8 @@
 from ccpi.viewer.CILViewerBase import CILViewerBase
 from ccpi.viewer.utils import colormaps
 
+from ccpi.viewer.CILViewerBase import CILViewerBase
+
```
Comment (lauramurgatroyd) [inline Wrappers/Python/ccpi/viewer/CILViewer.py:26]:
```suggestion
```

Comment (lauramurgatroyd) [reply Wrappers/Python/ccpi/viewer/CILViewer.py:26]: repeated import

File: Wrappers/Python/ccpi/web_viewer/dev-environment.yml [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
 name: cilviewer_webapp
+channels:
+  - conda-forge
+  - ccpi
 dependencies:
 - python==3.9
 - matplotlib # Optional for more colormaps
 - h5py
 - pip
+- ccpi::cil-data
```
Comment (lauramurgatroyd) [inline Wrappers/Python/ccpi/web_viewer/dev-environment.yml:11]: does this force installing from the ccpi channel? If so, that is very useful!!

File: Wrappers/Python/ccpi/viewer/CILViewer.py (hunk 1 above)
Comment (Pasarus) [reply Wrappers/Python/ccpi/viewer/CILViewer.py:26]: Good spot!

File: Wrappers/Python/ccpi/web_viewer/dev-environment.yml (hunk 2 above)
Comment (Pasarus) [reply Wrappers/Python/ccpi/web_viewer/dev-environment.yml:11]: Yes! it should do that

File: .github/workflows/docker_build_test_publish.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Build docker container
+        shell: bash -l {0}
+        run: |
+          docker build -t cil-viewer docker/web-app
+
+      - name: Run docker container with tests
+        shell: bash -l {0}
+        run: |
+          docker run --rm --entrypoint /bin/bash -v /home/runner/work/CILViewer/CILViewer:/root/source_code  cil-viewer -c "source ./mambaforge/etc/profile.d/conda.sh && conda activate cilviewer_webapp && conda install cil-data pytest -c ccpi && python -m pytest /root/source_code/Wrappers/Python -k 'not test_version'"
+    # TODO: publish to come later
```
[the remark this answers is not in the document]
Comment (lauramurgatroyd) [reply .github/workflows/docker_build_test_publish.yml:37]: ah but it's not in the docker environment file! is there a reason why?

[the remark this answers is not in the document]
Comment (Pasarus) [reply .github/workflows/docker_build_test_publish.yml:37]: It's not needed in the file.

[the remark this answers is not in the document]
Comment (Pasarus) [reply .github/workflows/docker_build_test_publish.yml:37]: https://github.com/vais-ral/CILViewer/issues/285

**Pull request merged**